### PR TITLE
Make function to branches references into compressed or uncompressed

### PR DESCRIPTION
**Pull request by Lucpen**
## PR checklist

- [x] Make function to branches eferences into compressed or uncompressed
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/genomic-medicine-sweden/tomte/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/prepare_references.nf [hunk 1]
```diff
@@ -171,3 +124,29 @@ workflow PREPARE_REFERENCES {
         vep_cache     = ch_final_vep.collect()                 // channel: [ path(cache) ]
         versions      = ch_versions                            // channel: [ path(versions.yml) ]
 }
+// Custom functions
+/**
+* Branch a channel into differnt channels,
```
Comment (Lucpen) [inline subworkflows/local/prepare_references.nf:129]:
```suggestion
* Branch a channel into diffrent channels,
```

Comment (Lucpen) [inline subworkflows/local/prepare_references.nf:129]:
```suggestion
* Branch a channel into different channels,
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,7 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - GATK4_ASEREADCOUNTER and GATK4_SPLITNCIGARREADS have been updated [#101](https://github.com/genomic-medicine-sweden/tomte/pull/101)
 - Updated GATK4_ASEREADCOUNTER, now bam and vcf will be given as one channel [#103](https://github.com/genomic-medicine-sweden/tomte/pull/103)
 - Prepare reference subworkflow has been reformated and simplified [#105](https://github.com/genomic-medicine-sweden/tomte/pull/105)
-- FastQC have been updated to correctly allocate memory [#168](https://github.com/genomic-medicine-sweden/tomte/pull/168)
+- FastQC have been updated to correctly allocate memory [#106](https://github.com/genomic-medicine-sweden/tomte/pull/106)
```
Comment (fellen31) [inline CHANGELOG.md:20]: Thank you :)

**Review by fellen31 — APPROVED** [review]
LGTM! ⚡

File: subworkflows/local/prepare_references.nf [hunk 3]
```diff
@@ -34,14 +34,7 @@ workflow PREPARE_REFERENCES {
...
         GUNZIP_FASTA(ch_fasta)
-        ch_fasta.
-            branch{ meta, fasta ->
-                compressed: fasta.toUriString().endsWith(".gz") // If the file ends with .gz
-                    return [ meta, fasta ]
-                uncompressed: !(fasta.toUriString().endsWith(".gz")) // If the file doesn't end with .gz
-                    return [ meta, fasta ]
-                }
-                .set{ch_fasta_mix}
+        ch_fasta_mix = branchChannelToCompressedAndUncompressed(ch_fasta)
```
Comment (jemten) [inline subworkflows/local/prepare_references.nf:37]: Neat!!

**Review by jemten — APPROVED** [review]
So clean 🤩

**Pull request merged**